playbooks/openstack-osh_hostconfig.yml

```diff
@@ -0,0 +1,59 @@
+---
+- hosts: osh-deployer
+  gather_facts: no
+  tasks:
+    - name: Ensure all the vars are loaded
+      include_vars: "{{ playbook_dir }}/../vars/deploy-on-openstack.yml"
+
+    - name: Configure host repositories
+      raw: |
+        zypper ar -f -G {{ deploy_on_openstack_repos_to_configure[item] }} {{ item }}
+      loop:
+        - SLE12SP3-product
+        - SLE12SP3-update
+        - SLE12SP3SDK-product
+        - SLE12SP3SDK-update
+        - SUSE-CA
+        - OpenStack-Cloud8-product
+        - SLE12Containers
+
+    - name: Refresh zypper. After this step, we can rely on ansible modules.
+      raw: |
+        #TODO(evrardjp): Check if utilities/devel tools/virt/cloud8updates are really needed
+        #zypper ar -f -G https://provo-mirror.opensuse.org/repositories/utilities/SLE_12/ SLE12-utilities;
+        #zypper ar -f -G https://provo-mirror.opensuse.org/repositories/Virtualization:/containers/SLE_12_SP3/ Virtualization_containers;
+        #zypper ar -f -G  http://kirk.provo.novell.com/repositories/SUSE/Products/OpenStack-Cloud/8/x86_64/product/ openstackcloud-8-update
+        zypper refresh
+
+    - name: Upgrade host
+      zypper:
+        state: dist-upgrade
+        name: '*'
+
+    #On some images this service is not defined and the systemctl command fails.
+    #The failed_when false is temporary until I find out proper conditional on this task.
+    - name: Disable firewall
+      systemd:
+        enabled: no
+        name: "{{ item }}"
+      failed_when: false
+      with_items:
+        - SuSEfirewall2_setup.service
+        - SuSEfirewall2_init.service
+
+    - name: Prepare keys
+      shell: |
```

Comment:
cant we use `openssh_keypair` to generate the keys instead of shell? Also this seems liek 2 different tasks merged into one

Reply:
We should indeed :)